### PR TITLE
Add update command for refreshing all channel files

### PR DESCRIPTION
**Pull request by j16r**
Add an update command, which iterates over every channel and update's its Youtube details. If you specify a slug it will only update the channel matching the slug.

Import command now only introduces a new channel, erroring if a channel file already exists.

[Finishes #88]

Examples:

## New channel:

```
bake channel import sarcasmatron youtube https://www.youtube.com/channel/UC6WD1VImCVeOuNccuqDzAdA
2019/03/03 15:11:48 Importing https://www.youtube.com/channel/UC6WD1VImCVeOuNccuqDzAdA...
2019/03/03 15:11:50 Title: Sarcasmitron, Count: 2178
2019/03/03 15:11:50 Saving ../data/channels/sarcasmatron.yml
```

## Existing channel:

```
bake channel import sarcasmatron youtube https://www.youtube.com/channel/UC6WD1VImCVeOuNccuqDzAdA
2019/03/03 15:12:20 Importing https://www.youtube.com/channel/UC6WD1VImCVeOuNccuqDzAdA...
2019/03/03 15:12:23 Channel with slug 'sarcasmatron' already exists, use update instead.
```

## Update without any arguments: 

```
bake channel update
2019/03/03 15:12:45 Updating channels...
2019/03/03 15:12:47 Saving ../data/channels/friendlyjordies.yml
2019/03/03 15:12:47 Saving ../data/channels/grahamelwood.yml
2019/03/03 15:12:47 Saving ../data/channels/knowingbetter.yml
...
```

## Update with some arguments:

```
bake channel update friendlyjordies angiespeaks typomaster
2019/03/03 15:13:28 Updating channels [friendlyjordies angiespeaks typomaster]...
2019/03/03 15:13:29 Saving ../data/channels/friendlyjordies.yml
2019/03/03 15:13:30 Saving ../data/channels/angiespeaks.yml
2019/03/03 15:13:30 Couldn't find channel with slug 'typomaster', skipping...
```

### CODE REVIEW

**Comment by quietlemon** [comment]
I briefly looked at the code, it looks nice!

I would suggest renaming `YouTubeURL` function to something else (`MainURL`? `MainProviderURL`?) for future-proofing reasons (i.e. supporting other platforms). Instead of looking up the `youtube` key by default, maybe it could extract the first key from the provider map that comes up? (which usually will be youtube)

**Comment by jaymickey** [comment]
@quietlemon Definitely a valid concern, and I think it's definitely worth further consideration, especially around making the update command provider agnostic - I would recommend raising an issue!

Having said that in the spirit of delivery over quality the current implementation should be sufficient. As it currently stands we pretty much only have an implementation for YouTube as a provider, and further discussion around how we handle additional video providers would likely need to happen!

File: data/channels/theserfs.yml [hunk 1]
```diff
@@ -9,11 +9,11 @@ providers:
...
     subscribers: 0
   youtube:
-    name: The Serfs
+    name: ""
     slug: ""
-    url: https://youtube.com/theserfstv
+    url: null
     description: ""
-    subscribers: 5483
+    subscribers: 0
```
Comment (dirkkelly) [inline data/channels/theserfs.yml]: data should not be cleared

File: data/channels/erincollective.yml [hunk 2]
```diff
@@ -1,4 +1,11 @@
 name: Erin Collective
+providers:
```
Comment (dirkkelly) [inline data/channels/erincollective.yml]: cleared out

**Pull request merged**